Develop

Sergej-Skom opened the pull request.

Sergej-Skom committed:
add jacoco report

Sergej-Skom committed:
The project is assembled, the coverage is one hundred percent

Sergej-Skom committed:
The project is assembled, the coverage is one hundred percent

digranate commented on src/test/java/com/example/LionTest.java:
```diff
@@ -0,0 +1,58 @@
+@RunWith(Parameterized.class)
+public class LionTest {
+    private final String sex;
+    private final boolean havMane;
```
have

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы. Применены моки и параметризация 
Замечания:
1. Убрать TestCase 
2. Убрать дублирование тестов 
3. Нейминг

Sergej-Skom committed:
Inheritance has been removed

Sergej-Skom committed:
1. Testing was performed via "assert Throws".
2. The twice-repeated test has been removed, since it lies in a parameterized class.
3. The name of the method from "getCredentials" has been renamed to "getLionTestCases".
4. All "havMane" are changed to "haveMane".

digranate commented on src/test/java/com/example/CatTest.java:
```diff
@@ -0,0 +1,31 @@
+    public void testGetFood() throws Exception {
+        when(feline.eatMeat()).thenReturn(List.of("one", "two"));
+        var cat = new Cat(feline);
+        assertEquals(List.of("one", "two"), cat.getFood());
```
Дважды создаешь один и тот же список.
Можно улучшить

pomegranate282922 reviewed (commented):
Привет! Многое исправил - супер! Но куда пропал тест на исключение в Lion? Его нужно вернуть, но так, чтобы он не повторялся дважды. 
также не вижу тестов на getKittens у feline(( 
И тест на исключение в Cat тоже выполнится два раза, так как лежит в параметризованном классе. 
прошу замечания применить ко всему(!) коду.

Sergej-Skom committed:
1. Lion's exception test is called "testInvalidSexThrowsException".
2. feline's "getKittens" is added.
3. The exception test in "Cat" has been fixed.
4. The comments are applied to the entire code.

digranate commented on src/main/java/com/example/Feline.java:
```diff
@@ -15,11 +15,10 @@ public String getFamily() {
 
     public int getKittens() {
-        return getKittens(1);
+        return 1;
```
Зачем это ? Мы не меняем ничего в исходном коде. Кроме изоляции в Lion.
Нужно написать тесты на существующий (!) Код. 

Нужно исправить

digranate commented on src/test/java/com/example/LionTest.java:
```diff
@@ -41,6 +42,12 @@ public void testGetKittens() throws Exception {
+    }
+
+    @Test
+    public void testInvalidSexThrowsException() {
```
Так. Еще раз: этот ( и предыдущий) тест повторится дважды, так как лежит в параметризованном классе. Смысла в том чтобы он выполнится дважды - нет. Поэтому это нужно исправить.

pomegranate282922 reviewed (commented):
Привет! Есть ощущение, что не до конца понято замечание про дублирование теста - написана развернутее. 
также не надо менять исходный код кроме изоляции - но на тестах это изменение не сказывается ( поэтому вдвойне непонятно зачем оно).

Sergej-Skom committed:
Redone

digranate commented on src/test/java/com/example/CatTest.java:
```diff
@@ -1,33 +1,36 @@
+        feline.eatMeat();
+        Mockito.verify(feline, Mockito.times(1)).eatMeat();
+        Mockito.when(feline.eatMeat()).thenReturn(expectedEatMeat);
+        Assert.assertEquals(expectedEatMeat, feline.eatMeat());
```
Это ассерт НЕ к месту - мы здесь тестируем метод класса Cat. Проверять что замокированный метод вернет ожидаемый результат - это тестирование библиотеки мокито- это НЕ нужно.

digranate commented on src/test/java/com/example/FelineTest.java:
```diff
@@ -1,30 +1,61 @@
+        assertEquals(expectedKittensCount, actualKittensCount);
     }
-}
+}
```
Каждый тест в этом классе повторится 3 раза, так как они лежат в параметризованном классе. Так быть не должно. 
нужно исправить.

digranate commented on src/test/java/com/example/LionTest.java:
```diff
@@ -1,53 +1,68 @@
         assertEquals("Используйте допустимые значения пола животного - самец или самка", exception.getMessage());
     }
-}
+}
```
И здесь каждый тест повторится 2 раза - прошу это исправить уже третье ревью.

нужно исправить

pomegranate282922 reviewed (commented):
Привет! Еще раз пишу замечание: убрать повторение тестов из-за того что они все лежат в параметризованном классе. Это нужно исправить.

Sergej-Skom committed:
Changed

Sergej-Skom committed:
A new class, depending on the lion from the feline

pomegranate282922 reviewed (approved):
Ок, зачет!

Sergej-Skom commented:
Merge
